src/gu/sceGuLight.c

```diff
@@ -10,15 +10,21 @@
 
 void sceGuLight(int light, int type, int components, const ScePspFVector3 *position)
 {
-	GuLightSettings* settings = &light_settings[light];
+	int offset = light * 3;
+	int ctype;
 
-	sendCommandf(settings->xpos,position->x);
-	sendCommandf(settings->ypos,position->y);
-	sendCommandf(settings->zpos,position->z);
+	if (components == GU_POWERED_DIFFUSE) {
+		ctype = GU_DIFFUSE_AND_SPECULAR;
+	} else if (components == GU_DIFFUSE_AND_SPECULAR) {
+		ctype = GU_POINTLIGHT;
+	} else {
+		ctype = GU_DIRECTIONAL;
+	}
 
-	int kind = 2;
-	if (components != 8)
-		kind = (components^6) < 1 ? 1 : 0;
-
-	sendCommandi(settings->type,((type & 0x03) << 8)|kind);
+	// Light position
+	sendCommandf(LIGHT0_X + offset, position->x);
+	sendCommandf(LIGHT0_Y + offset, position->y);
+	sendCommandf(LIGHT0_Z + offset, position->z);
+	// Light type and components
+	sendCommandi(LIGHT_TYPE0 + light, (ctype << 8) | (type));
```

Comment:
I think this should be like this for it to match what we had before:
```suggestion
	sendCommandi(LIGHT_TYPE0 + light, ((type & GU_AMBIENT_AND_DIFFUSE) << 8) | ctype);
```

Reply:
I see, then it is probably best to keep it in case the user gives invalid input. Especially since it was allowed before.